OpenACC for simpleince and modsurface(lmostlocal=True)

fjansson opened the pull request:
Add OpenACC directives to simpleice - seems to work, not seriously tested

Also add one missing directive in modsurface - gives a weird warning
`NVFORTRAN-W-1054-Module variables used in acc routine need to be in acc declare create() `
and then fails when linking with `Undefined reference to 'modsurface_patchynr_' in 'CMakeFiles/dales.dir/modsurface.f90.o'` @CasparJungbacker any idea how to fix that? Otherwise we can live without patch support for now.

fjansson committed:
modsimpleice: add OpenACC directives

CasparJungbacker commented on src/modsurface.f90:
```diff
@@ -1675,6 +1675,7 @@ function factorial(k)
   function patchxnr(xpos)
+    !$acc routine seq
     use modmpi,     only : myidx
     use modglobal,  only : imax,itot
```
I think the issue here is that `patchxnr` and `patchynr` use global variables. Hence, these need to be copied to the GPU explicitly by using `!$acc declare create`. So for example, we would need to place an `!$acc declare create(imax, itot)` near the declaration of both of them. That being said, I think the more elegant solution here is to pass everything as arguments to the functions.

fjansson replied:
I added !$acc declare create() for all the variables needed, to mess as little as possible with the patch{x,y}nr functions.

fjansson committed:
modsurface: add OpenACC directives for lmostlocal=True

fjansson committed:
modsurface and modibm: use only the local thvsl in the Obukhov solver

The local Obukhov length solver (lmostlocal = .true.) used both the global thvs and
a locally calculated value thvsl. Now use only the locally calculated value.
thvs caused problems on GPU, as it was !$acc declare'd but not actually updated.
